go-controller/pkg/node/node_smartnic.go

```diff
@@ -0,0 +1,203 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+//watchSmartNicPods watch updates for pod smart nic annotations
+func (n *OvnNode) watchSmartNicPods() {
+	var retryPods sync.Map
+	// servedPods tracks the pods that got a VF
+	var servedPods sync.Map
+	_ = n.watchFactory.AddPodHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod := obj.(*kapi.Pod)
+			klog.Infof("AddFunc for POD: %s/%s", pod.ObjectMeta.GetNamespace(), pod.ObjectMeta.GetName())
+			if !util.PodWantsNetwork(pod) || pod.Status.Phase == kapi.PodRunning {
+				return
+			}
+			if util.PodScheduled(pod) {
```

Comment:
do we still need this check I thought we listen on pod on the same node?

Reply:
it actually depends on the type of watch factory (master or node),  only the node watch factory limits listening on pods on a specific node.

looking at `ovnkube.go` there is a mode where both master and node run in the same process, in this case the the master watch factory will be used which does not limit the scope to a specific node.

https://github.com/ovn-org/ovn-kubernetes/blob/0a2aad51333400e28fbf51c1017b3064d61186cc/go-controller/cmd/ovnkube/ovnkube.go#L266


IMO it is safer to not assume filtration is performed at informer level.

Reply:
Ok make sense